server/asset/eth/eth.go

```diff
@@ -202,8 +218,17 @@ func (eth *Backend) BlockChannel(size int) <-chan *asset.BlockUpdate {
 }
 
 // Contract is part of the asset.Backend interface.
-func (eth *Backend) Contract(coinID []byte, redeemScript []byte) (*asset.Contract, error) {
-	return nil, notImplementedErr
+func (eth *Backend) Contract(coinID, _ []byte) (*asset.Contract, error) {
+	sc, err := newSwapCoin(eth, coinID, sctInit)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create coiner: %v", err)
+	}
```

Comment:
Maybe should check Confirmations once for errors here, in case it is already mined.